app/main.py

```diff
@@ -132,6 +132,20 @@ async def delete(request):
     status = await (dqueue.cancel(ids) if where == 'queue' else dqueue.clear(ids))
     return web.Response(text=serializer.encode(status))
 
+@routes.get(config.URL_PREFIX + 'history')
+async def list_queue(_):
+    history = [];
+    for _ ,v in dqueue.queue.saved_items():
+        obj = vars(v)
+        obj['list_type'] = 'queue'
+        history.append(obj)
+    for _ ,v in dqueue.done.saved_items():
+        obj = vars(v)
+        obj['list_type'] = 'done'
+        history.append(obj)
+
+    return web.Response(text=json.dumps(history))

```

Comment:
If you restructure the response as suggested above, then you don't need to add a property to the object (`list_type`), and then you can call `serializer.encode(history)` here instead of `json.dumps(history)`, and if you do that, you don't need to convert the objects to dicts above with `vars()`.. you can probably just use the return value of `save_items()` directly, it'll make the code much shorter

Reply:
Thanks for your tips, i'll make the changes. I have one question, metube does't store the error message in the queue does it? if i restart the container the list is still there, however the error become null